python/ray/train/tests/test_gpu.py

```diff
@@ -62,6 +62,8 @@ def train_fn():
         assert devices == [0, 0]
     elif num_gpus_per_worker == 1:
         assert devices == [0, 1]
+    elif num_gpus_per_worker == 2:
+        assert devices == [0]
```

Comment:
Does this run? Might need to increase the fixture to have 4 GPUs since there are 2 workers?

Also this probably doesn't test complete coverage (I think the previous code would succeed here as well).

Would be good to have a test where the index is not equal to the GPU ID.